### PR TITLE
ICMSLST-536 Add country groups data migration.

### PR DESCRIPTION
**Pull request by trbateman**

### CODE REVIEW

File: web/domains/country/models.py [hunk 1]
```diff
@@ -16,29 +16,24 @@ class Country(models.Model):
...
-            return "(new)"
+        return self.name
 
     @property
     def name_slug(self):
         return self.name.lower().replace(" ", "_")
 
     class Meta:
         ordering = ("name",)
+        unique_together = ["name", "is_active"]
```
Comment (oskusalerma) [inline web/domains/country/models.py:27]: unique_together is deprecated: https://docs.djangoproject.com/en/3.1/ref/models/options/#unique-together

use UniqueConstraint instead. see web/domains/template/models.py for an example

File: web/domains/country/models.py [hunk 2]
```diff
@@ -16,29 +16,24 @@ class Country(models.Model):
     hmrc_code = models.CharField(max_length=20, blank=False, null=False)
 
     def __str__(self):
-        if self.id:
```
Comment (oskusalerma) [inline web/domains/country/models.py:19]: why are you changing this?

File: web/domains/country/models.py [hunk 3]
```diff
@@ -16,29 +16,24 @@ class Country(models.Model):
...
     class Meta:
         ordering = ("name",)
+        unique_together = ["name", "is_active"]
 
 
 class CountryGroup(models.Model):
-    name = models.CharField(max_length=4000, blank=False, null=False)
-    comments = models.CharField(max_length=4000, blank=True, null=True)
+    name = models.CharField(max_length=4000, blank=False, null=False, unique=True)
+    comments = models.TextField(blank=True, null=True)
```
Comment (oskusalerma) [inline web/domains/country/models.py:32]: why are you changing this from charfield to textfield, and in effect, removing the max length?

File: web/domains/country/models.py [hunk 4]
```diff
@@ -16,29 +16,24 @@ class Country(models.Model):
...
     countries = models.ManyToManyField(Country, blank=True, related_name="country_groups")
 
     def __str__(self):
-        if self.id:
```
Comment (oskusalerma) [inline web/domains/country/models.py:38]: why are you changing this?

File: web/migrations/0027_add_country_data.py [hunk 5]
```diff
@@ -240,7 +240,7 @@ def load_country_data(apps, schema_editor):
                 hmrc_code="CG",
```
Comment (oskusalerma) [inline web/migrations/0027_add_country_data.py:240]: umm, migration 26 has been merged to develop, me and kevin have run it, you can't just change history now.

**Review by oskusalerma — CHANGES_REQUESTED** [review]

File: web/domains/country/models.py (hunk 2 above)
Comment (trbateman) [reply web/domains/country/models.py:19]: Returning name gives a more meaningful description of the object, especially in the admin.

File: web/domains/country/models.py (hunk 1 above)
Comment (trbateman) [reply web/domains/country/models.py:27]: Thanks for heads up. It's not deprecated at the moment but might be in the future, hence no deprecation warnings coming up.

File: web/domains/country/models.py (hunk 3 above)
Comment (trbateman) [reply web/domains/country/models.py:32]: Text field feels more appropriate given the string has line breaks in it. The default widget is therefore more suitable for django admin.

File: web/domains/country/models.py (hunk 4 above)
Comment (trbateman) [reply web/domains/country/models.py:38]: Same as above

File: web/migrations/0027_add_country_data.py (hunk 5 above)
Comment (trbateman) [reply web/migrations/0027_add_country_data.py:240]: Will give you a call about this.

**Review by oskusalerma — APPROVED** [review]

**Pull request merged**